labkey-client-api/src/org/labkey/remoteapi/Connection.java

```diff
@@ -125,73 +124,100 @@
-     * <code>http://localhost:8080/labkey</code>
+     * <code>new URI("http://localhost:8080/labkey")</code>
      * <p>
      * Note that https may also be used for the protocol. By default the
      * Connection is configured to deny self-signed SSL certificates.
      * If you want to accept self-signed certificates, use
      * <code>setAcceptSelfSignedCerts(false)</code> to enable this behavior.
      * </p>
-     * The email name and password should correspond to a valid user email
-     * and password on the target server.
-     * @param baseUrl The base URL
+     * @param baseURI Base URI for this Connection
      * @param credentialsProvider A credentials provider
      */
-    public Connection(String baseUrl, CredentialsProvider credentialsProvider)
+    public Connection(URI baseURI, CredentialsProvider credentialsProvider)
     {
-        try
+        if (baseURI.getHost() == null || baseURI.getScheme() == null)
         {
-            _baseURI = new URI(baseUrl);
+            throw new IllegalArgumentException("Invalid server URL: " + baseURI.toString());
         }
-        catch (URISyntaxException e)
-        {
-            throw new IllegalArgumentException("Invalid target server URL. " + baseUrl);
-        }
-        _baseUrl = baseUrl;
+        _baseURI = baseURI;
         _credentialsProvider = credentialsProvider;
         _httpClientContext = HttpClientContext.create();
         _httpClientContext.setCookieStore(new BasicCookieStore());
         setAcceptSelfSignedCerts(false);
     }
 
+    /**
+     * Constructs a new Connection object given a base URL and a credentials provider.
+     * <p>
+     * The baseUrl parameter should include the protocol, domain name, port,
+     * and LabKey web application context path (if configured). For example
+     * in a typical localhost configuration, the base URL would be:
+     * </p>
+     * <code>http://localhost:8080/labkey</code>
+     * <p>
+     * @param baseUrl The base URL
+     * @param credentialsProvider A credentials provider
+     * @see #Connection(URI, CredentialsProvider)
+     */
+    public Connection(String baseUrl, CredentialsProvider credentialsProvider)
+    {
+        this(toURI(baseUrl), credentialsProvider);
+    }
+
     /**
      * Constructs a new Connection object with a base URL that attempts authentication via .netrc/_netrc entry, if present.
      * If not present, connects as guest.
      * @param baseUrl The base URL
      * @throws URISyntaxException if the given url is not a valid URI
      * @throws IOException if there are problems reading the credentials
-     * @see #Connection(String, CredentialsProvider)
+     * @see NetrcCredentialsProvider
+     * @see #Connection(URI, CredentialsProvider)
      */
     public Connection(String baseUrl) throws URISyntaxException, IOException
     {
-        this(baseUrl, new NetrcCredentialsProvider(new URI(baseUrl)));
+        this(new URI(baseUrl), new NetrcCredentialsProvider(new URI(baseUrl)));
```

Comment:
I would just call `toURI()` here (both places) for consistency. I know you're following the previous impl, but it's hard to justify some constructors throwing `IllegalArgumentException` and others throwing `URISyntaxException`.

Reply:
Ok. I left the unused `throws URISyntaxException` to maintain backward compatibility.